ui-spacetimechart/src/utils/canvas.ts

```diff
@@ -2,54 +2,90 @@ import { clamp, identity } from 'lodash';
 
 import type { Direction, PathEnd, Point, RGBAColor, RGBColor } from '../lib/types';
 
+/**
+ * This function returns the picking layers scaling ratio. We basically take the min of the screen
+ * pixels and the "HTML pixels", and divide it by two.
+ *
+ * This allows having a smaller picking stage to fill (so it's faster), while keeping a "good enough
+ * precision".
+ */
+export function getPickingScalingRatio(): number {
+  const PICKING_DOWNSCALING_RATIO = 0.5;
+  const dpr = window.devicePixelRatio || 1;
+
+  // When devicePixelRatio is over 1 (like for Retina displays), we downscale based on the "HTML
+  // pixels":
+  if (dpr > 1) return PICKING_DOWNSCALING_RATIO;
+
+  // When devicePixelRatio is under 1 (like when the user zooms out for instance), we downscale
+  // based on the actual "screen pixels" (to avoid having a too large scene to fill):
+  if (dpr < 1) return PICKING_DOWNSCALING_RATIO * dpr;
+
+  return PICKING_DOWNSCALING_RATIO;
```

Comment:
```suggestion
  // When devicePixelRatio is over 1 (like for Retina displays), we downscale based on the "HTML
  // pixels":
  if (dpr >= 1) return PICKING_DOWNSCALING_RATIO;

  // When devicePixelRatio is under or equal to 1 (like when the user zooms out for instance), we downscale
  // based on the actual "screen pixels" (to avoid having a too large scene to fill):
  return PICKING_DOWNSCALING_RATIO * dpr;
```

Reply:
I just integrated that modification, thanks!